driver-core/src/main/com/mongodb/internal/async/AsyncRunnable.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.async;
+
+import com.mongodb.internal.async.function.RetryState;
+import com.mongodb.internal.async.function.RetryingAsyncCallbackSupplier;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * See AsyncRunnableTest for usage
+ */
+public interface AsyncRunnable {
+
+    static AsyncRunnable startAsync() {
+        return (c) -> c.onResult(null, null);
+    }
```

Comment:
[optional]

`startAsync` implies starting something asynchronously, e.g., submitting a task to be executed in the background. Consider something like `newAsyncRunnable`.

Reply:
Let's try `beginAsync`